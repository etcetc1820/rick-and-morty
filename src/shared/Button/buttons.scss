button {
  display: block;
  margin: 10px auto 0;
  padding: 5px 25px;
  cursor: pointer;
}
