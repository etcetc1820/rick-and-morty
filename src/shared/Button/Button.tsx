import React from "react";

import "./buttons.scss";

interface ButtonProps {
  disabled?: boolean;
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  ...props
}): JSX.Element => {
  return (
    <button {...props} onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
