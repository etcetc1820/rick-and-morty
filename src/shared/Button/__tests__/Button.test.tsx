import React from "react";
import rendered from "react-test-renderer";
import { cleanup, render } from "@testing-library/react";

import Button from "../Button";

afterEach(cleanup);

const handlerClick = (): void => {
  console.log("do smth");
};

describe("Button", () => {
  it("button without text", () => {
    const buttonComponent = rendered.create(<Button />);
    const tree = buttonComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("button with text", () => {
    const buttonComponent = rendered.create(<Button>Hopa</Button>);
    const tree = buttonComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("button with text and function", () => {
    const buttonComponent = rendered.create(
      <Button onClick={handlerClick}>Hopa</Button>
    );
    const tree = buttonComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("disabled button", () => {
    const { getByText } = render(
      <Button onClick={handlerClick} disabled>
        Hopa
      </Button>
    );

    expect(getByText(/Hopa/)).toHaveAttribute("disabled");
  });
});
