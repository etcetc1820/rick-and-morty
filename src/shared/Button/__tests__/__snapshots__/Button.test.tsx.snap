// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button with text 1`] = `
<button>
  Hopa
</button>
`;

exports[`Button button with text and function 1`] = `
<button
  onClick={[Function]}
>
  Hopa
</button>
`;

exports[`Button button without text 1`] = `<button />`;
